### PR TITLE
TOOL-21808 Install Java 17 on DCoL hosts

### PR DESCRIPTION
**Pull request by dmittal-delphix**
<details open>
<summary><h2> Problem </h2></summary>
Jenkins controller migration to java17 requires all agents to be on java17
</details>

<details open>
<summary><h2> Solution </h2></summary>

Install java17 on the dcol systems
</details>

<details>
<summary><h2> Testing Done </h2></summary>
For testing first these changes need to be merged then appliance build would be run . This is based on my understanding as controller deployment fails with this on dev.

```
2024-12-03 22:41:27,356 - ERROR   - Key 'appliance-build' is not one of the supported repositories: ['devops-gate', 'pipeline-shared', 'linux-pkg', 'qa-infra']
Traceback (most recent call last):
  File "/Users/deepali.mittal/git-repos/jenkins_java17/devops-gate/jm/lib/python3.12/site-packages/dlpx/jenkins/manager/cli.py", line 381, in main
    repos = list(
            ^^^^^
  File "/Users/deepali.mittal/git-repos/jenkins_java17/devops-gate/jm/lib/python3.12/site-packages/dlpx/jenkins/manager/repos_parser.py", line 44, in parse_git_repos
    raise exceptions.JenkinsManagerException(
dlpx.jenkins.manager.exceptions.JenkinsManagerException: Key 'appliance-build' is not one of the supported repositories: ['devops-gate', 'pipeline-shared', 'linux-pkg', 'qa-infra']
```

https://selfservice-jenkins.eng-tools-prd.aws.delphixcloud.com/job/appliance-build-orchestrator-pre-push/9945/console
</details>

### CODE REVIEW

File: live-build/variants/internal-dcenter/package-lists/dcenter.list.chroot [hunk 1]
```diff
@@ -24,6 +24,7 @@ libldap2-dev
 libsasl2-dev
 nfs-kernel-server
 openjdk-11-jdk-headless
+openjdk-17-jdk-headless
```
Comment (sebroy) [inline live-build/variants/internal-dcenter/package-lists/dcenter.list.chroot:27]: Do we need both Java 11 and Java 17 on dcol?

**Review by grodr — APPROVED** [review]

File: live-build/variants/internal-dcenter/package-lists/dcenter.list.chroot
Comment (grodr) [inline live-build/variants/internal-dcenter/package-lists/dcenter.list.chroot:1]: For testing you should be able to kick off appliance-build-orchestrator on the selfservice Jenkins instance with your PR branch as the APPLIANCE_BUILD_GIT_BRANCH https://selfservice-jenkins.eng-tools-prd.aws.delphixcloud.com/job/appliance-build-orchestrator-pre-push/build?delay=0sec I think the only variant that you need is `internal-dcenter` for VARIANTS and IMPORT_VARIANTS

Comment (grodr) [reply live-build/variants/internal-dcenter/package-lists/dcenter.list.chroot:1]: PLATFORMS should only require `esx` as well.

**Review by sebroy — APPROVED** [review]

File: live-build/variants/internal-dcenter/package-lists/dcenter.list.chroot
Comment (dmittal-delphix) [reply live-build/variants/internal-dcenter/package-lists/dcenter.list.chroot:1]: Thanks Garret .. The job was successful . 
https://selfservice-jenkins.eng-tools-prd.aws.delphixcloud.com/job/appliance-build-orchestrator-pre-push/9945/console

File: live-build/variants/internal-dcenter/package-lists/dcenter.list.chroot (hunk 1 above)
Comment (dmittal-delphix) [reply live-build/variants/internal-dcenter/package-lists/dcenter.list.chroot:27]: yes dcoa build still uses java11 , for jenkins use we are installing java17

**Pull request merged**